.github/workflows/az-deploy.yml

```diff
@@ -0,0 +1,37 @@
+name: Deploy Azure Resources
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v3
+
+      - name: Setup Azure CLI
+        run: |
+          curl -sL https://aka.ms/InstallAzureCLIDeb | sudo bash
+          az --version  # Verify installation
+
+      - name: Login to Azure
+        run: |
+          az login --service-principal -u ${{ secrets.AUTO_AZURE_CLIENT_ID }} -p ${{ secrets.AUTO_AZURE_CLIENT_SECRET }} --tenant ${{ secrets.AUTO_AZURE_TENANT_ID }}
+          az account set --subscription ${{ secrets.AUTO_AZURE_SUBSCRIPTION_ID }}
+
+      - name: Install Bicep CLI
+        run: az bicep install
+
+      - name: Deploy Bicep Template
+        id: deploy
+        run: |
+          set -e
+          az deployment sub create \
+            --name automation-CWYD \
+            --location eastus \
+            --template-file infra/main.bicep \
+            --parameters environmentName=pslautomationtest location=eastus2
```

Comment:
Should there also be a step to tear this down and clean up the resources?

Reply:
yes, we are implementing that also.